src/UndisclosedPassword.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+
+namespace Zend\Validator;
+
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+
+final class UndisclosedPassword extends AbstractValidator
+{
+    const HIBP_API_URI = 'https://api.pwnedpasswords.com';
+    const HIBP_API_TIMEOUT = 300;
+    const HIBP_CLIENT_UA = 'zend-validator';
+    const HIBP_CLIENT_ACCEPT = 'application/vnd.haveibeenpwned.v2+json';
+    const HIBP_RANGE_LENGTH = 5;
+    const HIBP_RANGE_BASE = 0;
+    const HIBP_COUNT_BASE = 0;
+    const SHA1_LENGTH = 40;
+
+    const PASSWORD_BREACHED = 'passwordBreached';
+    const WRONG_INPUT = 'wrongInput';
+    const CONNECTION_FAILURE = 'connFail';
+    const UNKNOWN_ERROR = 'unknownError';
+
+    protected $messageTemplates = [
+        self::PASSWORD_BREACHED => 'The provided password was used by others',
+        self::WRONG_INPUT => 'The provided password failed to be correctly hashed, please verify your input',
+        self::CONNECTION_FAILURE => 'Unable to reach HIBP service, please try again later',
+        self::UNKNOWN_ERROR => 'Something happened beyond our control, error reporting should give more details',
+    ];
+
+    /**
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $httpRequest;
+
+    /**
+     * @var ResponseFactoryInterface
+     */
+    private $httpResponse;
+
+    /**
+     * @var int
+     */
+    private $count = 0;
```

Comment:
What count is this representing? Looking just at the property, this can be confusing

Reply:
Count is used for providing a "count" of amount of times a password was found in Have I been pwned. Each hash that returns contains a number. See https://haveibeenpwned.com/API/v2#SearchingPwnedPasswordsByRange for details.

Reply:
> amount of times a password was found

`pwnedPasswordsFoundCount`?

Reply:
Or I could implement `Countable` so you could count on the object directly.

Reply:
I don't think that would clarify it: it would make it even more confusing instead. Why would a `Countable` implemented on a `ValidatorInterface` implementation returning a "count of pwned found passwords"? This is just about communicating what a specific `private` state is meant for, so we shouldn't expand the `public` API of the object for an implementation detail that is already ambiguous in `private` scope.

Reply:
Good point, in a "validation" context it has no real purpose. I'll remove it.